src/display/Display.spec.js

```diff
@@ -1 +1,52 @@
-// Test away!
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { render, cleanup } from 'react-testing-library';
+import 'jest-dom/extend-expect';
+import Display from './Display';
+afterEach(cleanup);
+
+describe('<Display />', () => {
+  it('Should render without crashing', () => {
+    render(<Display />);
+  });
+  it('Should match snapshot', () => {
+    const tree = renderer.create(<Display />);
+    expect(tree.toJSON()).toMatchSnapshot();
+  });
+  it('Should display gate as open by default', () => {
+    const { getByText } = render(<Display />);
+    getByText(/Open/i);
+  });
+  it('Should display gate as unlocked by default', () => {
+    const { getByText } = render(<Display />);
+    getByText(/Unlocked/i);
+  });
+  it('Should display gate as closed if closed prop is passed', () => {
+    const { getByText } = render(<Display closed />);
+    getByText(/Closed/i);
+  });
+  it('Should display gate as locked if locked prop is passed', () => {
+    const { getByText } = render(<Display locked />);
+    getByText(/^Locked/i);
```

Comment:
You could make helper functions to create the similar test functions; for example:
```javascript
const renderWith = (component, text) => () => {
  const { getByText } = render(component);
  getByText(text);
};
```
and for the tests below:
```javascript
const renderWithClass = (component, text, className) => {
  const { queryByText } = render(component);
  const status = queryByText(text);
  expect(status).toHaveClass(className);
};
```

Reply:
their usages, using the test above and below, would be as follows:
```javascript
it('Should display gate as locked if locked prop is passed', 
  renderWith(<Display locked />, /^Locked/i)
);

it('Should use red-led class when locked',
  renderWithClass(<Display locked />, /^Locked/i, 'red-led')
);
```

Reply:
You could also potentially use these helpers in other test files (which is largely why I wrote them to take in a component rather than just `Display`'s props)